packages/svelte/src/reactivity/map.js

```diff
@@ -41,18 +41,24 @@ export class ReactiveMap extends Map {
 
 	/** @param {K} key */
 	has(key) {
-		var s = this.#sources.get(key);
+		var sources = this.#sources;
+		var s = sources.get(key);
 
 		if (s === undefined) {
-			// We should always track the version in case
-			// the Set ever gets this value in the future.
-			get(this.#version);
-
-			return false;
+			// If we're working with a non-primitive then we can generate a signal for it
+			if (typeof key !== 'object' || key === null) {
+				s = source(UNINITIALIZED);
+				sources.set(key, s);
```

Comment:
This still has the same problem as #11504 and #11287. We need a way to remove these signals when we are no longer listening to them. Otherwise it would create a memory leak (see https://github.com/sveltejs/svelte/pull/11504#issuecomment-2129529254)

Reply:
Fixed. Unfortunately, this means that we can't retain primitives either. So they won't be fine-grain, as there's no way to do so without leaking.